docs/mdbook/src/android/location-component-guide.md

```diff
@@ -0,0 +1,101 @@
+# LocationComponent
+
+This guide will demonstrate how to utilize the [LocationComponent] to represent the user's current location.
+
+
+1. When implementing the [LocationComponent], the application should request location permissions.
+
+   Declare the need for foreground location in the `AndroidManifest.xml` file.
+
+   For more information, please refer to the [Android Developer Documentation].
+
+```xml
+<manifest ... >
+  <!-- Always include this permission -->
+  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+
+  <!-- Include only if your app benefits from precise location access. -->
+  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+</manifest>
+```
+
+2. Create a new activity named `BasicLocationPulsingCircleActivity`:
+  - This Activity Implements the `OnMapReadyCallback` interface. The `onMapReady()` method is used to manage the application once the map is ready.
+  - Add a variable `permissionsManager` to manage permissions.
+  - Add a variable `locationComponent` to manage user location.
+  - At the end of the `onCreate()` method, call `checkPermissions(`) to ensure that the application can access the user's location.
+
+```kotlin
+{{#include ../../../../platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/BasicLocationPulsingCircleActivity.kt:top}}
+```
+
+3. In the `checkPermissions()` method, the [PermissionManager] is used to request location permissions at runtime and handle the callbacks for permission granting or rejection.
+
+   Additionally, you should pass the results of `Activity.onRequestPermissionResult()` to it.
+
+   If the permissions are granted, call `mapView.getMapAsync(this)` to execute the `onMapReady()` method when the map is ready.
```

Comment:
I would go with:

If the permissions are granted, call `mapView.getMapAsync(this)` to register the activity as a listener for `onMapReady` event.

Reply:
Done in cfda02a